File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();
        for(char c : s.toCharArray()){
            if(c == ')'){
                if(stk.isEmpty() || stk.pop() != '(') return false;
            }else if(c == ']'){
                if(stk.isEmpty() || stk.pop() != '[') return false;
            }else if(c == '}'){
                if(stk.isEmpty() || stk.pop() != '{') return false;
            }else{
                stk.add(c);
            }
        }
        return stk.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stk = new Stack();
    for(let i = 0; i < s.length; i++){
        if(s[i] === ')'){
            if(stk.pop() != '(') return false;
        }else if(s[i] === ']'){
            if(stk.pop() != '[') return false;
        }else if(s[i] === '}'){
            if(stk.pop() != '{') return false;
        }else{
            stk.push(s[i]);
        }
    }
    return stk.isEmpty();
};